### PR TITLE
Table key

### PR DESCRIPTION
**Pull request by sivask94**
Error in “var TableRows” and “var TableHeader”

### CODE REVIEW

File: public/scripts/dummy.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
+var headers = [
```
Comment (Sharavanth) [inline public/scripts/dummy.js:1]: Why is this file committed..... If ur using thi, it needs to be included in index.html

File: public/scripts/Table.js [hunk 2]
```diff
@@ -17,9 +39,17 @@ var TableHeader = React.createClass({
     render: function () {
         return (
             <tr>
-                {this.props.head.map(function (data, idx) {
-                    return <HeaderCell data={data}/>;
-                })}
+                {
+
+                    for(var p in obj1)
```
Comment (Sharavanth) [inline public/scripts/Table.js]: `obj1` is the data that needs to be passed to the TableHeader React Class......
If so, it must be made available through props......

the error u must be getting is `undefined` one, ryt?...

**Review by Sharavanth — CHANGES_REQUESTED** [review]

File: public/scripts/Table.js (hunk 2 above)
Comment (sivask94) [reply public/scripts/Table.js]: That was one error and also looping error sir "Expected } error"

File: public/scripts/Table.js [hunk 3]
```diff
@@ -1,29 +1,31 @@
-var headers = {
+var headers = [
 
-    Name: 'DisplayName',
-    Quantity: 'Quantity in Kgs',
-    Price: 'Price in Rs'
-};
+{Name: 'DisplayName'},
```
Comment (Sharavanth) [inline public/scripts/Table.js]: indentation needs to be corrected

File: public/scripts/Table.js [hunk 4]
```diff
@@ -36,18 +38,22 @@ var HeaderCell = React.createClass({
 });
 
 var TableHeader = React.createClass({
-    getHeaderCells: function () {
-        for (var p in headers) {
-            if (headers.hasOwnProperty(p))
-            return <HeaderCell data={headers[p]}/>;
+    getHeaderCells: function (heads) {
+        var cell=[];
```
Comment (Sharavanth) [inline public/scripts/Table.js]: could be named as `cells`..... (since it is an array)

File: public/scripts/Table.js [hunk 5]
```diff
@@ -36,18 +38,22 @@ var HeaderCell = React.createClass({
...
+        for (var p in heads) {
+            if (heads.hasOwnProperty(p))
+                cell.push(<HeaderCell data={heads[p]}/>);
         }
+        return cell;
     },
     render: function () {
-
+        var sel=this;
```
Comment (Sharavanth) [inline public/scripts/Table.js]: should be named as self

File: public/scripts/Table.js [hunk 6]
```diff
@@ -63,27 +69,29 @@ var Rowcell = React.createClass({
...
 
 var TableRows = React.createClass({
-    getRowkeys: function (row) {
-        for(var k in row) {
-            if (row.hasOwnProperty(k))
-                return <Rowcell cell={row[k]}/>;
 
+    getRowKeys: function (row) {
+        var cells = [];
+        for (var k in row) {
```
Comment (Sharavanth) [inline public/scripts/Table.js]: this loop is run only on the `data` object..... So no `header keys` from `headers` are referred here....

this is why the result table does not have proper cells in the proper columns

```
headers.map(function(header, headerKey){
 for (var k in row) {
  if (row.hasOwnProperty(k))
   cells.push(<Rowcell cell={row[headerKey]}/>);
 }
});
```

**Review by Sharavanth — CHANGES_REQUESTED** [review]

**Review by Sharavanth — APPROVED** [review]

**Pull request merged**